File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第8章快速幂.py
#快速幂 lanqiaoOJ 1514

def fastPow(a,n,mod):
    ans = 1
    while n:
        if(n&1):  ans = ans*a 
        a = a*a 
        n >>= 1
    return ans % mod

b,p,k = map(int, input().split())
print(fastPow(b,p,k))

File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第8章快速幂.cpp
//快速幂
//lanqiaoOJ 1514

#include<bits/stdc++.h>
using namespace std;
typedef long long ll;                      //变量改用较大的long long
ll fastPow(ll a, ll n, ll mod){
    ll ans = 1;
    a %= mod;                             //重要，防止下面的ans*a越界
    while(n) {
        if(n & 1)   ans = (ans*a) % mod;   //取模
        a = a*a % mod;                     //取模
        n >>= 1;
    }
    return ans;
}
int main(){
    ll b,p,k;    cin>>b>>p>>k;
    cout << fastPow(b,p,k);
    return 0;
}
